### PR TITLE
`Programming exercises`: Adjust access rights for submission policies

### PR DESCRIPTION
**Pull request by niclasschuemann**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
This PR addresses some UI/UX improvements regarding the submission policy feature that are leftover from the main PR. The concrete issues and changes are listed in the **Description** section.

### Description
1. When creating programming exercises, the _Generate_ button is now disabled when the submission policy configuration form is invalid
**Now:**
submission-policy-generate-exercise-button-disabled

**Previously:**
submission-policy-generate-exercise-button-enabled

2. Submission policy form is disabled for editors on the grading page
3. Editors can no longer use the add and update submission policy functionality (server side change)
**Now:**
submission-policy-editor-visibility

**Previously:**
submission-policy-editor-visibility-previously

4. When reloading the submission policy grading page, the submission policy grading page is now loaded and not the test case grading page.
5. The release and saved changes indicator is no longer visible on the submission policy grading page
6. On the submission policy grading page, the _Update_ button is now disabled when the submission policy configuration form is invalid
**Now:**
submission-policy-updated-grading

**Previously:**
submission-policy-previous-grading

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Editor

Discussed in testing session

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Coverage
submission-policy-update.component.ts: 100% -> 98%

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ import {
 } from 'app/exercises/programming/manage/services/programming-exercise-grading.service';
 import { SubmissionPolicyService } from 'app/exercises/programming/manage/services/submission-policy.service';
 import { SubmissionPolicy, SubmissionPolicyType } from 'app/entities/submission-policy.model';
+import { SubmissionPolicyUpdateComponent } from 'app/exercises/shared/submission-policy/submission-policy-update.component';
```
Comment (MDK5) [inline src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts:26]: !

File: src/test/javascript/spec/component/shared/submission-policy-update.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        // We revert the enum values, since 'none' is the default type for the type picker. Therefore we
+        // need to set the picker to anything other than 'none' to test switching back to 'none' appropriately
+        for (const type of Object.values(SubmissionPolicyType).reverse()) {
+            submissionPolicyTypeField.value = type;
+            submissionPolicyTypeField.dispatchEvent(new Event('change'));
+            fixture.detectChanges();
+            tick();
+
+            expect(expectedProgrammingExercise.submissionPolicy?.type).toBe(type);
+            expect(expectedProgrammingExercise.submissionPolicy?.id).toBeUndefined();
```
Comment (MDK5) [inline src/test/javascript/spec/component/shared/submission-policy-update.component.spec.ts:52]:
```suggestion
            expect(expectedProgrammingExercise.submissionPolicy?.id).toBe(undefined);
```
according to our guidelines

**Review by MDK5 — CHANGES_REQUESTED** [review]
Tested in testing session, the described improvements worked, good job :+1:

Also reviewed code, changes look good to me, just two nitpicks

File: src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts (hunk 1 above)
Comment (niclasschuemann) [reply src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts]: Removed

File: src/test/javascript/spec/component/shared/submission-policy-update.component.spec.ts (hunk 2 above)
Comment (niclasschuemann) [reply src/test/javascript/spec/component/shared/submission-policy-update.component.spec.ts]: Fixed

**Review by Malyuk-A — APPROVED** [review]
Tested on ts1 during testing session. Works as described 👍

**Review by MDK5 — APPROVED** [review]
Comments got addressed, reapprove

**Pull request merged**